packages/hardhat-core/test/internal/hardhat-network/provider/HardhatBlockchain.ts

```diff
@@ -146,6 +176,34 @@ describe("HardhatBlockchain", () => {
       assert.equal(await blockchain.getBlock(blockThree.hash()), undefined);
     });
 
+    it.skip("removes the block and all subsequent ones (callbackified)", async () => {
```

Comment:
This test is a duplicated version of the previous one, like the other added tests. But this one fails because of this line:

https://github.com/nomiclabs/hardhat/blob/7b606216090859ac0edbca105d6cbd49113c209a/packages/hardhat-core/src/internal/hardhat-network/provider/types/PBlockchain.ts#L34

I don't know if that line is fine, and so this tests needs to be different, or if that undefined check should be removed